packages/@aws-cdk/aws-appmesh/lib/virtual-node-listener.ts

```diff
@@ -0,0 +1,348 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+import { validateHealthChecks } from './private/utils';
+import { HealthCheck, HttpTimeout, TcpTimeout, Protocol, GrpcTimeout } from './shared-interfaces';
+
+/**
+ * Properties for a VirtualNode listener
+ */
+export interface VirtualNodeListenerConfig {
+  /**
+   * Single listener config for a VirtualNode
+   */
+  readonly listener: CfnVirtualNode.ListenerProperty,
+}
+
+/**
+ *  Defines listener for a VirtualNode
+ */
+export abstract class VirtualNodeListener {
+  /**
+   * Returns an HTTP Listener for a VirtualNode
+   */
+  public static httpNodeListener(props: HttpNodeListenerProps = {}): VirtualNodeListener {
+    return new HttpNodeListener(props);
+  }
+
+  /**
+   * Returns an HTTP2 Listener for a VirtualNode
+   */
+  public static http2NodeListener(props: HttpNodeListenerProps = {}): VirtualNodeListener {
```

Comment:
Can you move these `*Props` interfaces above the `VirtualNodeListener` class please?

Also, we generally call them `Props` only when they're used for Construct properties. For static methods like these, they should have the suffix 'Options' (so, `HttpNodeListenerOptions`).

Reply:
These `*NodeListenerProps` are being used to initialize the constructor of the `*NodeListener` classes as the static method returns `*NodeListener` object and the props are used to initialize the constructor. Wouldn't it be okay to call them `*Props` here ?

Reply:
No, because `VirtualNodeListener` is not a construct. We use `*Props` only for construct properties.